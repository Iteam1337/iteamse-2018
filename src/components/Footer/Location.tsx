import React from 'react'
import { FooterQuery } from '../../../typings/iteamse'
import styled from '../../theme'
import UnstyledList from '../List/UnstyledList'

interface LocationProps {
  address: FooterQuery['addresses'][0]
}

const Wrap = styled.div`
  color: #000;
`

const Title = styled.div`
  font-weight: 500;
  margin-bottom: 20px;
`

const Address = styled.div`
  margin-bottom: 20px;
`

const Location: React.SFC<LocationProps> = ({ address }) => {
  return (
    <Wrap key={address.zip}>
      <Title>{address.title}</Title>
      <Address>
        <UnstyledList>
          <li>{address.address1}</li>
          <li>
            {address.zip} {address.city}
          </li>
        </UnstyledList>
      </Address>
      <UnstyledList>
        <li>{address.contactPhone}</li>
        <li>{address.contactMail}</li>
        <li>ORG: {address.orgNumber}</li>
      </UnstyledList>
    </Wrap>
  )
}

export default Location
