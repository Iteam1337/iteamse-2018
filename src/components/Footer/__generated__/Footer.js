

/* @flow */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: Footer
// ====================================================

export type Footer_addresses = {
  __typename: "Address",
  address1: string,
  city: string,
  contactMail: string,
  contactPhone: string,
  orgNumber: string,
  title: string,
  zip: string,
};

export type Footer = {
  addresses: Array<Footer_addresses>
};

//==============================================================
// START Enums and Input Objects
// All enums and input objects are included in every output file
// for now, but this will be changed soon.
// TODO: Link to issue to fix this.
//==============================================================

//==============================================================
// END Enums and Input Objects
//==============================================================