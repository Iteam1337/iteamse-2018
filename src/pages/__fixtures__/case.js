export const workCase = {
  headerBgColor: '#1616B2',
  logo:
    '//images.ctfassets.net/rj4r6yfcesw5/4p0ptdLj9SaECekOI4c6C2/c2cdeab1cebc9650ebb0e7778357c231/AF_ordbild_vit.png',
  slug: 'arbetsformedlingen',
  thumbnailImage:
    '//images.ctfassets.net/rj4r6yfcesw5/3Y7zZ25WbeOCU8w84kAEos/b97ff1899cf6f2e4dca0eb2b4f0d5a44/af.png',
  tags: ['API', 'Docker', 'UX', 'React', 'Vue'],
  title: 'Arbetsförmedlingen',
  introduction:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  introductionTitle: 'En ny digital tjänst',
  process:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  processTitle: 'Process',
  development:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  developmentTitle: 'Utveckling',
  aboutCompany:
    'Aeneanurutrum, erat at tempus porttitor, mi eros vestibulum magna, eget semper mauris ipsum a nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Cras quis odio vel turpis pellentesque finibus. Pellentesque cursus diam ac ligula placerat volutpat. Vivamus dictum augue eu rutrum efficitur. Suspendisse ut pellentesque ante. Suspendisse odio velit, fringilla eu porta vel, elementum ac libero. Nam eu vestibulum magna. Vestibulum quis risus elementum, lobortis sapien at, rhoncus enim. Nam accumsan odio lacus, at facilisis lacus rhoncus vel. Suspendisse sollicitudin a ipsum at cursus.',
  aboutCompanyTitle: 'Arbetsförmedlingen',
  partners: '[Digitalbyrån](https://hej.com)',
  partnersTitle: 'Samarbetspartners',
  contact:
    '- [Hans Rollman](mailto:hans.rollman@iteam.se)\n- Johan Littorin, Telenor, 0734 41 72 85',
  contactTitle: 'Presskontakt',
  quote:
    '"We feel that our crew at Iteam are more colleagues than suppliers. Rather than specifying a tech spec to work on, we collaborate, carefully chiseling out functionality as we go along. This way of working has minimized waste and it has also taken us further than we ever imagined”',
  quotePerson: 'Anna Andersson, chef Arbetsförmedlingen',
  __typename: 'Case',
}
