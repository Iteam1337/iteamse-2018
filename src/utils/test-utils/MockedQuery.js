import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import ApolloClient from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import createMockedNetworkFetch from 'apollo-mocknetworkinterface'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

type P = {
  children: React.Node,
  loading?: boolean,
  response: Object,
}

const MockedQuery = ({ children, loading, response = {} }: P) => {
  const createResponse = () => {
    return {
      loading: !!loading,
      data: {
        ...response,
      },
    }
  }

  const mockedNetworkFetch = createMockedNetworkFetch(createResponse, {
    latency: 0,
  })

  const client = new ApolloClient({
    link: createHttpLink({
      uri: 'http://localhost:3000',
      fetch: mockedNetworkFetch,
    }),
    cache: new InMemoryCache(),
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>{children}</BrowserRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
