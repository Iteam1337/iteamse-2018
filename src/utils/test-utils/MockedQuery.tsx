import { InMemoryCache } from 'apollo-cache-inmemory'
import ApolloClient from 'apollo-client'
import * as React from 'react'
import { ApolloProvider } from 'react-apollo'
import { mockSingleLink } from 'react-apollo/test-utils'
import MockRouter from 'react-mock-router'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../theme'

interface MockedQueryProps {
  location?: object
  mocks: any[]
  params?: object
}

const MockedQuery: React.SFC<MockedQueryProps> = ({
  children,
  location,
  mocks,
  params,
}) => {
  const link = mockSingleLink.apply(null, mocks)

  const client = new ApolloClient({
    cache: new InMemoryCache(),
    link,
  })

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <MockRouter location={location} params={params}>
          {children}
        </MockRouter>
      </ThemeProvider>
    </ApolloProvider>
  )
}

export default MockedQuery
