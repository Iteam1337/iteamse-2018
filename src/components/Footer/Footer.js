// @flow

import React from 'react'
import * as Iteam from '../../typings/iteam.flow'
import * as IteamCMS from './__generated__/Footer'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import Location from './Location'
import GridRow from '../Grid/GridRow'
import styled from 'styled-components'

type Props = Iteam.ApolloBase<IteamCMS.Footer>

const FooterQuery = gql`
  query Footer {
    addresses {
      address1
      city
      contactMail
      contactPhone
      orgNumber
      title
      zip
    }
  }
`

const Wrap = GridRow.extend`
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 40px;
  padding-top: 40px;

  @media (min-width: 1024px) {
    padding-left: 0;
    padding-right: 0;
  }
`

const Locations = styled.div`
  display: grid;
  grid-gap: 30px;

  @media (min-width: 1024px) {
    grid-template-columns: repeat(2, 315px);
  }
`

const Footer = () => {
  return (
    <Query query={FooterQuery}>
      {({ loading, data: { addresses } }: Props) => {
        if (loading) {
          return null
        }

        return (
          <Wrap>
            <Locations>
              {addresses.map(address => (
                <Location key={address.zip} address={address} />
              ))}
            </Locations>
          </Wrap>
        )
      }}
    </Query>
  )
}

export default Footer
