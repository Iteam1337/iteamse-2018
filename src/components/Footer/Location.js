// @flow

import React from 'react'
import * as IteamCMS from './__generated__/Footer'
import styled from 'styled-components'

type Props = {
  address: IteamCMS.Footer_addresses,
}

const Wrap = styled.div`
  color: #000;

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }
`

const Title = styled.div`
  font-weight: 500;
  margin-bottom: 20px;
`

const Address = styled.div`
  margin-bottom: 20px;
`

const Location = ({ address }: Props) => {
  return (
    <Wrap key={address.zip}>
      <Title>{address.title}</Title>
      <Address>
        <ul>
          <li>{address.address1}</li>
          <li>
            {address.zip} {address.city}
          </li>
        </ul>
      </Address>
      <ul>
        <li>{address.contactPhone}</li>
        <li>{address.contactMail}</li>
        <li>ORG: {address.orgNumber}</li>
      </ul>
    </Wrap>
  )
}

export default Location
