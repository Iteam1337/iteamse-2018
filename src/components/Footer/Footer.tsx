import gql from 'graphql-tag'
import React from 'react'
import { Query } from 'react-apollo'
import styled from 'styled-components'
import { FooterQuery } from '../../../typings/iteamse'
import { OPERATIONS_PAGE_QUERY } from '../../pages/Ops'
import GridColumn from '../Grid/GridColumn'
import NativeLink from '../Link/NativeLink'
import StyledPrefetchLink from '../Link/StyledPrefetchLink'
import UnstyledList from '../List/UnstyledList'
import iconFb from './img/icon_fb.svg'
import iconInstagram from './img/icon_instagram.svg'
import iconLinkedin from './img/icon_linkedin.svg'
import iconMynewsdesk from './img/icon_mynewsdesk.svg'
import iconYoutube from './img/icon_youtube.svg'
import Location from './Location'

export const FOOTER_QUERY = gql`
  query Footer {
    addresses {
      address1
      city
      contactMail
      contactPhone
      orgNumber
      title
      zip
    }
  }
`

const Wrap = GridColumn.extend`
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 40px;
  padding-top: 40px;

  @media (min-width: 1025px) {
    grid-row-gap: 40px;
    padding-left: 0;
    padding-right: 0;
  }
`

const FooterSections = styled.div`
  display: grid;
  grid-gap: 30px;

  @media (min-width: 1025px) {
    grid-template-columns: repeat(2, 315px) 1fr;
  }

  /* IE 11 */
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    width: 100%;
    max-width: 1024px;
    display: flex;
    justify-content: space-around;

    > * {
      flex: 1 0 25%;
    }
  }
`

const Title = styled.div`
  font-weight: 500;
  margin-bottom: 20px;
`

const SocialMedia = styled.div``
const SocialMediaItem = styled.li`
  align-items: center;
  display: flex;
  margin-bottom: 10px;
`
const SocialMediaIcon = styled.img`
  margin-right: 15px;
`

const NativeLinkUnderlined = NativeLink.extend`
  border-bottom: 2px solid;
  color: black;
  text-decoration: none;
`

const SupportSection = styled.div`
  /* IE 11 */
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    display: flex;
    width: 100%;
  }
`

class FooterQueryComponent extends Query<FooterQuery> {}

const Footer = () => {
  return (
    <FooterQueryComponent query={FOOTER_QUERY}>
      {({ loading, data }) => {
        if (loading || !data) {
          return null
        }

        const { addresses } = data

        return (
          <Wrap>
            <FooterSections>
              {addresses.map(address => (
                <Location key={address.zip} address={address} />
              ))}

              <SocialMedia>
                <Title>Följ oss</Title>
                <UnstyledList>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconFb} alt="" />
                    <NativeLinkUnderlined href="https://www.facebook.com/iteam1337/">
                      Facebook
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconInstagram} alt="" />
                    <NativeLinkUnderlined href="https://instagram.com/iteam1337">
                      Instagram
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconLinkedin} alt="" />
                    <NativeLinkUnderlined href="https://www.linkedin.com/company/270569/">
                      LinkedIn
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconYoutube} alt="" />
                    <NativeLinkUnderlined href="https://www.youtube.com/channel/UCU2TdLZ7p0jEuuGUxaod4lw">
                      YouTube
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                  <SocialMediaItem>
                    <SocialMediaIcon src={iconMynewsdesk} alt="" />
                    <NativeLinkUnderlined href="http://www.mynewsdesk.com/se/iteam">
                      MyNewsDesk
                    </NativeLinkUnderlined>
                  </SocialMediaItem>
                </UnstyledList>
              </SocialMedia>
            </FooterSections>
            <SupportSection>
              <StyledPrefetchLink
                black={true}
                query={OPERATIONS_PAGE_QUERY}
                to="ops"
              >
                Behöver du supporthjälp?
              </StyledPrefetchLink>
            </SupportSection>
          </Wrap>
        )
      }}
    </FooterQueryComponent>
  )
}

export default Footer
