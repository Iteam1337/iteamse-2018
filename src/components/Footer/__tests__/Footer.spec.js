import React from 'react'
import Footer from '../Footer'
import { render, wait } from 'react-testing-library'
import MockedQuery from '../../../utils/test-utils/MockedQuery'
import { addresses } from '../__fixtures__/addresses'

describe('components/Footer', () => {
  const mockedResponse = {
    addresses,
  }

  it('renders Footer', async () => {
    const { getByText, container } = render(
      <MockedQuery response={mockedResponse}>
        <Footer />
      </MockedQuery>
    )

    await wait(() => getByText('Stockholm'))

    expect(container).toMatchSnapshot()
  })
})
