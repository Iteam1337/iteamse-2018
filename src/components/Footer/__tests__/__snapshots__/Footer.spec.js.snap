// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Footer renders Footer 1`] = `
<div>
  <div
    class="GridColumn__GridColumnClean-lcQAdD iWPUQR"
  >
    <div
      class="Footer__FooterSections-xtla5m-0 jWuJXj"
    >
      <div
        class="Location__Wrap-s1et3i43-0 gDiDVe"
      >
        <div
          class="Location__Title-s1et3i43-1 hOnjvR"
        >
          Iteam Göteborg
        </div>
        <div
          class="Location__Address-s1et3i43-2 bVugku"
        >
          <ul
            class="UnstyledList-s1ntpurm-0 gEDRFE"
          >
            <li>
              Haga Kyrkogata 28
            </li>
            <li>
              411 23
              Göteborg
            </li>
          </ul>
        </div>
        <ul
          class="UnstyledList-s1ntpurm-0 gEDRFE"
        >
          <li>
            +46 734 029 112
          </li>
          <li>
            johanna.grahn@iteam.se
          </li>
          <li>
            ORG: 
            556551-6928
          </li>
        </ul>
      </div>
      <div
        class="Location__Wrap-s1et3i43-0 gDiDVe"
      >
        <div
          class="Location__Title-s1et3i43-1 hOnjvR"
        >
          Iteam Stockholm
        </div>
        <div
          class="Location__Address-s1et3i43-2 bVugku"
        >
          <ul
            class="UnstyledList-s1ntpurm-0 gEDRFE"
          >
            <li>
              Östermalmsgatan 26A
            </li>
            <li>
              114 26
              Stockholm
            </li>
          </ul>
        </div>
        <ul
          class="UnstyledList-s1ntpurm-0 gEDRFE"
        >
          <li>
            +46 826 70 90
          </li>
          <li>
            info@iteam.se
          </li>
          <li>
            ORG: 
            556551-6928
          </li>
        </ul>
      </div>
      <div
        class="Footer__SocialMedia-xtla5m-2 fltlKn"
      >
        <div
          class="Footer__Title-xtla5m-1 hAElLH"
        >
          Följ oss
        </div>
        <ul
          class="UnstyledList-s1ntpurm-0 gEDRFE"
        >
          <li
            class="Footer__SocialMediaItem-xtla5m-3 eUeiui"
          >
            <img
              class="Footer__SocialMediaIcon-xtla5m-4 bRvQuN"
              src="icon_fb.svg"
              alt=""
            />
            <a
              class="NativeLink-s1qo6rwq-0 jNinQt"
              href="https://www.facebook.com/iteam1337/"
            >
              Facebook
            </a>
          </li>
          <li
            class="Footer__SocialMediaItem-xtla5m-3 eUeiui"
          >
            <img
              class="Footer__SocialMediaIcon-xtla5m-4 bRvQuN"
              src="icon_instagram.svg"
              alt=""
            />
            <a
              class="NativeLink-s1qo6rwq-0 jNinQt"
              href="https://instagram.com/iteam1337"
            >
              Instagram
            </a>
          </li>
          <li
            class="Footer__SocialMediaItem-xtla5m-3 eUeiui"
          >
            <img
              class="Footer__SocialMediaIcon-xtla5m-4 bRvQuN"
              src="icon_linkedin.svg"
              alt=""
            />
            <a
              class="NativeLink-s1qo6rwq-0 jNinQt"
              href="https://www.linkedin.com/company/270569/"
            >
              LinkedIn
            </a>
          </li>
          <li
            class="Footer__SocialMediaItem-xtla5m-3 eUeiui"
          >
            <img
              class="Footer__SocialMediaIcon-xtla5m-4 bRvQuN"
              src="icon_youtube.svg"
              alt=""
            />
            <a
              class="NativeLink-s1qo6rwq-0 jNinQt"
              href="https://www.youtube.com/channel/UCU2TdLZ7p0jEuuGUxaod4lw"
            >
              YouTube
            </a>
          </li>
          <li
            class="Footer__SocialMediaItem-xtla5m-3 eUeiui"
          >
            <img
              class="Footer__SocialMediaIcon-xtla5m-4 bRvQuN"
              src="icon_mynewsdesk.svg"
              alt=""
            />
            <a
              class="NativeLink-s1qo6rwq-0 jNinQt"
              href="http://www.mynewsdesk.com/se/iteam"
            >
              MyNewsDesk
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
